packages/experiments/src/components/Button/Button.types.tsx

```diff
@@ -13,6 +13,7 @@ export type IButtonSlots = 'root' | 'stack' | 'text' | 'icon' | 'menuIcon';
 
 export interface IButtonProps extends IStyleableComponentProps<IButtonProps, IButtonStyles> {
   as?: keyof JSX.IntrinsicElements;
+  componentRef?: IRefObject<IButtonComponent>;
```

Comment:
Hmm, this isn't correct, but I can see how the naming is confusing. `IButtonComponent` should probably be called `IButtonComponentTypes` (and `IComponent` be called `IComponentTypes`) as they are defining types, not actual methods on the class. componentRef must be an interface implemented by the component. In this case, `Button` will eventually need to implement something like [IButton](https://github.com/OfficeDev/office-ui-fabric-react/blob/74739d7e4154985c44ecd024d535d3a0b3b96627/packages/office-ui-fabric-react/src/components/Button/Button.types.ts#L12) but can be `{}` for now.

Reply:
Or possibly better yet for each case, define an `interface IButton = {}` then pass `IButton` to `componentRef`.